lib/api/backend_patients_list_api.dart

```diff
@@ -0,0 +1,194 @@
+            address: element['Address'] ?? '',
+            formOfAddress: element['Form'] ?? '',
+            objectId: element['objectId'],
+            createdAt: DateTime.parse(element['createdAt']),
+            updatedAt: DateTime.parse(element['updatedAt']),
+          ),
+        );
+      }
+
+      for (dynamic element in responsePatientData) {
+        patientDataResults.add(
+          PatientData(
+            bodyHeight: element['BodyHeight']['estimateNumber'].toDouble(),
+            patientID: element['ID'],
+            caseNumber: element['CaseNumber'],
+            instKey: element['inst_key'],
+            objectId: element['objectId'],
+            patientObjectId: element['Patient']['objectId'],
+            doctorObjectId: element['Doctor']['objectId'],
+            createdAt: DateTime.parse(element['createdAt']),
+            updatedAt: DateTime.parse(element['updatedAt']),
+          ),
+        );
+      }
+
+      for (dynamic element in responsePatientProfile) {
+        patientProfileResults.add(
+          PatientProfile(
+            weightBMIEnabled: element['Weight_BMI'],
+            heartFrequencyEnabled: element['HeartRate'],
+            bloodPressureEnabled: element['BloodPressure'],
+            bloodSugarLevelsEnabled: element['BloodSugar'],
+            walkDistanceEnabled: element['WalkingDistance'],
+            sleepDurationEnabled: element['SleepDuration'],
+            sleepQualityEnabled: element['SISQS'],
+            painEnabled: element['Pain'],
+            phq4Enabled: element['PHQ4'],
+            medicationEnabled: element['Medication'],
+            therapyEnabled: element['Therapies'],
+            doctorsVisitEnabled: element['Stays'],
+            patientObjectId: element['Patient']['objectId'],
+            doctorObjectId: element['Doctor']['objectId'],
+            objectId: element['objectId'],
+            createdAt: DateTime.parse(element['createdAt']),
+            updatedAt: DateTime.parse(element['updatedAt']),
+          ),
+        );
+      }
+
+      for (Patient patientObject in patientResults) {
+        String? patientObjectId = patientObject.objectId;
+
+        PatientData? matchingPatientData = patientDataResults.firstWhereOrNull(
+          (PatientData patientDataObject) =>
+              patientDataObject.patientObjectId == patientObjectId,
+        );
+
+        PatientProfile? matchingPatientProfile =
+            patientProfileResults.firstWhereOrNull(
+          (PatientProfile patientProfileObject) =>
+              patientProfileObject.patientObjectId == patientObjectId,
+        );
+
+        if (matchingPatientData != null && matchingPatientProfile != null) {
+          objectList.add(
+            PatientsListElement(
+              patient: patientObject,
+              patientData: matchingPatientData,
+              patientProfile: matchingPatientProfile,
+              objectId: patientObject.objectId,
+            ),
+          );
+        }
+      }
```

Comment:
There are edge cases, that will be very very difficult to debug, when they occur, because you simply won't know, where the problem is coming from.
E.G. you are getting all data from all tables. But the data is limited and you won't get actually all the data from a certain point, if the table grows. Since a patient will only get added to objectList if it is returned with matching patientData and patientProfile, you have three points of failure, where the nurse or doctor won't be able to find a patient. Because with the current implementation you can't ever be sure, if all three data sets get returned or not.

But this implementation will suffice as long as the patients tables don't exceed a certain amount.